### PR TITLE
Feature permissions projects

### PR DESCRIPTION
**Pull request by tnagorra**
The request is dependent on https://github.com/the-deep/client/pull/240

### CODE REVIEW

File: src/views/Project/Details/General/index.js [hunk 1]
```diff
@@ -38,11 +38,13 @@ const propTypes = {
     projectServerData: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
     setErrorProjectDetails: PropTypes.func.isRequired,
     projectId: PropTypes.number.isRequired,
+    disabled: PropTypes.bool.isRequired,
```
Comment (AdityaKhatri) [inline src/views/Project/Details/General/index.js]: Required and default props both are in place.
How do you mean?

File: src/views/Project/Details/Regions/ProjectRegionDetail/index.js [hunk 2]
```diff
@@ -51,6 +51,7 @@ const propTypes = {
     setRegionDetailsErrors: PropTypes.func.isRequired,
     removeProjectRegion: PropTypes.func.isRequired,
     onRegionClone: PropTypes.func,
+    disabled: PropTypes.bool.isRequired,
```
Comment (AdityaKhatri) [inline src/views/Project/Details/Regions/ProjectRegionDetail/index.js]: Required and default props both are in place.
How do you mean?

File: src/views/Project/Details/Users/ProjectUserList/Actions/index.js [hunk 3]
```diff
@@ -51,6 +51,11 @@ const propTypes = {
     activeUser: PropTypes.shape({
         userId: PropTypes.number,
     }).isRequired,
+    disabled: PropTypes.bool.isRequired,
```
Comment (AdityaKhatri) [inline src/views/Project/Details/Users/ProjectUserList/Actions/index.js]: Required and default props both are in place.
How do you mean?

File: src/views/Project/Details/Users/ProjectUserList/index.js [hunk 4]
```diff
@@ -29,10 +29,12 @@ const propTypes = {
 
     // eslint-disable-next-line react/no-unused-prop-types
     projectId: PropTypes.number.isRequired,
+    disabled: PropTypes.bool.isRequired,
```
Comment (AdityaKhatri) [inline src/views/Project/Details/Users/ProjectUserList/index.js]: Required and default props both are in place.
How do you mean?

File: src/views/Project/Details/Users/ProjectUsergroupList/Actions.js [hunk 5]
```diff
@@ -18,6 +18,11 @@ const propTypes = {
         role: PropTypes.string,
     }).isRequired,
     removeUsergroupMembershipRequest: RequestPropType.isRequired,
+    disabled: PropTypes.bool.isRequired,
```
Comment (AdityaKhatri) [inline src/views/Project/Details/Users/ProjectUsergroupList/Actions.js]: Required and default props both are in place.
How do you mean?

File: src/views/Project/Details/Users/ProjectUsergroupList/index.js [hunk 6]
```diff
@@ -26,10 +26,12 @@ const propTypes = {
 
     // eslint-disable-next-line react/no-unused-prop-types
     projectId: PropTypes.number.isRequired,
+    disabled: PropTypes.bool.isRequired,
```
Comment (AdityaKhatri) [inline src/views/Project/Details/Users/ProjectUsergroupList/index.js]: Required and default props both are in place.
How do you mean?

File: src/views/Project/Details/Users/SearchList/index.js [hunk 7]
```diff
@@ -25,10 +25,12 @@ const propTypes = {
     searchInputValue: PropTypes.string.isRequired,
     onSearchInputChange: PropTypes.func.isRequired,
     userSearchRequest: RequestPropType.isRequired,
+    disabled: PropTypes.bool,
```
Comment (AdityaKhatri) [inline src/views/Project/Details/Users/SearchList/index.js]: Required and default props both are in place.
How do you mean?

File: src/views/Project/Details/Users/index.js [hunk 8]
```diff
@@ -24,10 +24,12 @@ const propTypes = {
     usersRequest: PropTypes.shape({
         pending: PropTypes.bool.isRequired,
     }).isRequired,
+    disabled: PropTypes.bool.isRequired,
```
Comment (AdityaKhatri) [inline src/views/Project/Details/Users/index.js]: Required and default props both are in place.
How do you mean?

File: src/views/Project/Details/WordCategories/index.js [hunk 9]
```diff
@@ -25,10 +25,12 @@ const propTypes = {
     wordCategoryList: PropTypes.array.isRequired, // eslint-disable-line react/forbid-prop-types
     projectDetails: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
     setWordCategoryList: PropTypes.func.isRequired,
+    disabled: PropTypes.bool.isRequired,
```
Comment (AdityaKhatri) [inline src/views/Project/Details/WordCategories/index.js]: Required and default props both are in place.
How do you mean?

File: src/views/Project/Details/General/index.js (hunk 1 above)
Comment (tnagorra) [reply src/views/Project/Details/General/index.js]: haha bhetayecha, tara farak pardaina
runtime ma isRequired cha bhane console ma warning aucha, tetti nai ho 

Malai afai lai khojna alchi lagirathyo :D

**Pull request merged**